app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -1,25 +1,22 @@
 # frozen_string_literal: true
 
 class OmniauthCallbacksController < Devise::OmniauthCallbacksController
-  def google_oauth2
-    if user.present?
-      user.accept_invitation! if invitation_token
-      upload_avatar(omniauth_avatar)
-
-      flash[:notice] = I18n.t('devise.omniauth_callbacks.success',
-                              kind: t('omniauth.google'))
-      sign_in_and_redirect @user, event: :authentication
-    else
-      redirect_to new_user_session_path, notice: t('omniauth.access_denied')
+  def find_auth_translation(provider)
```

Comment:
we shall have to make this method private to this controller.

Even better we can move  `Google` from `omniauth:` to `navigation:` in the translations YAML. In that way we can have a neat `t('navigation.#{provider}')` this implementation scales.